src/test/random_tests.cpp

```diff
@@ -130,4 +130,19 @@ BOOST_AUTO_TEST_CASE(shuffle_stat_test)
     BOOST_CHECK_EQUAL(sum, 12000);
 }
 
+BOOST_AUTO_TEST_CASE(GetPoissonRand)
+{
+    g_mock_deterministic_tests = true;
+
+    int64_t now = 5000;
+    int average_interval_seconds = 600;
+
+    auto poisson = ::GetPoissonRand(now, average_interval_seconds);
+    std::chrono::microseconds poisson_chrono = ::GetPoissonRand(std::chrono::microseconds{now}, std::chrono::seconds{average_interval_seconds});
+
+    BOOST_CHECK_EQUAL(poisson, poisson_chrono.count());
```

Comment:
I'd be surprised if this works? Just because the randomizer is made deterministic does not mean it will produce the same value repeatedly.